receiver/vcenterreceiver/metrics.go

```diff
@@ -238,13 +248,23 @@ func (v *vcenterMetricScraper) processVMPerformanceMetrics(info *perfSampleResul
 					v.mb.RecordVcenterVMDiskLatencyMaxDataPoint(pcommon.NewTimestampFromTime(si.Timestamp), nestedValue)
 				}
 			}
+			if idx >= 1 && val.Instance != m.Value[idx-1].Instance {
+				v.mb.EmitForResource(metadata.WithVcenterClusterName(clusterName),
+					metadata.WithVcenterHostName(hostname),
+					metadata.WithVcenterVMID(vmUUID),
+					metadata.WithVcenterVMName(vmMain.Name()),
+					metadata.WithVcenterSystemDeviceID(m.Value[idx-1].Instance))
+			}
```

Comment:
What is going on here? What are we checking for in the condition and why are we emitting a resource when we find it?

Reply:
Yeah just chiming in and saying that the approach @dehaansa suggests would make what you're trying to do here more readable @BominRahmani. I also would probably want to validate more on if there is a content in the Instance variable. I know at least particularly for the VM network metrics that the instance can actually be the VM name itself which is an auto aggregation of the rest of the network interfaces...

All that to say is that we should do our due diligence and make sure content on this change is validated and the `expected.yaml` is strictly looked at to ensure we know which of the performance metrics are going to be affected. 

<img width="1205" alt="image" src="https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/32067685/b1497a8b-a73f-42b1-9afb-cc03446ca092">